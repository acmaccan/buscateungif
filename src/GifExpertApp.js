import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

const GifExpertApp = ({ defaultCategories = ["Mononoke"] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2>BuscateUnGif</h2>
      <AddCategory setCategories={setCategories} />
      <hr />

      <ol>
        {categories.map((category) => (
          <GifGrid category={category} key={category} />
        ))}
      </ol>
    </>
  );
};

export default GifExpertApp;
