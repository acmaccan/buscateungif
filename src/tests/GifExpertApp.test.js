import React from "react";
import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe("Test GifExpertApp", () => {
  test("should render GifExpertApp correctly", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should first", () => {
    const categories = [
      "My neighbor Totoro",
      "Spirited Away",
      "Princess Mononoke",
    ];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
