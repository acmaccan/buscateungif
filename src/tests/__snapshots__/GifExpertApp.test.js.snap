// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifExpertApp should first 1`] = `
<Fragment>
  <h2>
    BuscateUnGif
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="My neighbor Totoro"
      key="My neighbor Totoro"
    />
    <GifGrid
      category="Spirited Away"
      key="Spirited Away"
    />
    <GifGrid
      category="Princess Mononoke"
      key="Princess Mononoke"
    />
  </ol>
</Fragment>
`;

exports[`Test GifExpertApp should render GifExpertApp correctly 1`] = `
<Fragment>
  <h2>
    BuscateUnGif
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Mononoke"
      key="Mononoke"
    />
  </ol>
</Fragment>
`;
